bridge/client/app/_views/ktb-environment-view/ktb-environment-view.component.ts

```diff
@@ -1,29 +1,86 @@
+import { Stage } from '../../_models/stage';
+import { Location } from '@angular/common';
+
+export interface ISelectedStageInfo {
+  stage: Stage;
+  filterType: ServiceFilterType;
+}
 
 @Component({
   selector: 'ktb-environment-view',
   templateUrl: './ktb-environment-view.component.html',
   styleUrls: ['./ktb-environment-view.component.scss'],
   preserveWhitespaces: false,
 })
-export class KtbEnvironmentViewComponent {
+export class KtbEnvironmentViewComponent implements OnDestroy {
   @HostBinding('class') cls = 'ktb-environment-view';
-  public project$: Observable<Project | undefined>;
+  public project$: Observable<Project>;
+  private readonly unsubscribe$ = new Subject<void>();
+  public selectedStageInfo?: ISelectedStageInfo;
+
+  constructor(
+    private dataService: DataService,
+    private route: ActivatedRoute,
+    private router: Router,
+    private location: Location,
+    @Inject(POLLING_INTERVAL_MILLIS) private initialDelayMillis: number
+  ) {
+    const selectedStageName$ = this.route.paramMap.pipe(
+      map((params) => params.get('stageName')),
+      takeUntil(this.unsubscribe$)
+    );
 
-  constructor(private dataService: DataService, private route: ActivatedRoute) {
+    const paramFilterType$ = this.route.queryParamMap.pipe(map((params) => params.get('filterType')));
     const projectName$ = this.route.paramMap.pipe(
       map((params) => params.get('projectName')),
-      filter((projectName): projectName is string => !!projectName)
+      filter((projectName): projectName is string => !!projectName),
+      distinctUntilChanged()
     );
 
     this.project$ = projectName$.pipe(
       switchMap((projectName) => this.dataService.getProject(projectName)),
-      map((project) => (project?.projectDetailsLoaded ? project : undefined))
+      filter((project): project is Project => !!project?.projectDetailsLoaded)
+    );
+
+    combineLatest([selectedStageName$, paramFilterType$, this.project$])
+      .pipe(takeUntil(this.unsubscribe$))
+      .subscribe(([stageName, filterType, project]) => {
+        const stage = project.stages.find((s) => s.stageName === stageName);
+        if (stage) {
+          this.selectedStageInfo = { stage: stage, filterType: (filterType as ServiceFilterType) ?? undefined };
+        }
+      });
+
+    const projectTimer$ = AppUtils.createTimer(0, initialDelayMillis).pipe(
+      switchMap(() => projectName$),
+      takeUntil(this.unsubscribe$)
     );
+
+    projectTimer$.subscribe((projectName) => {
+      this.dataService.loadProject(projectName);
+    });
+  }
+
+  public setLocation(projectName: string, stageInfo: ISelectedStageInfo): void {
+    const url = this.router.createUrlTree(
+      ['/project', projectName, 'environment', 'stage', stageInfo.stage.stageName],
+      {
+        queryParams: { filterType: stageInfo.filterType },
+      }
+    );
+    this.location.go(url.toString());
```

Comment:
Because the whole environment view is re-rendered then and this would also trigger another API call to the project endpoint. This would happen every time a stage tile is clicked

Reply:
After discussion IRL we decided to keep it in as is and create a follow-up ticket for the router/routing/re-rendering situation.